.pre-commit-config.yaml

```diff
@@ -0,0 +1,18 @@
+repos:
+  # Using this mirror lets us use mypyc-compiled black, which is about 2x faster
+  - repo: https://github.com/psf/black-pre-commit-mirror
+    rev: 23.9.1
```

Comment:
Does this mean we're now pinning versions?

Reply:
Good question. I did not try to see if we can omit the rev. @jackvreeken If you agree that we should proceed with pre-commit, I'll give that a try and report back. 

Also, what do you think about using the GitHub action? I don't think I can implement that with my current level of access in the Pymcoa organization. If we don't use it on GitHub, then we can partially automate the local black fix ala https://github.com/pymoca/pymoca/pull/296#issuecomment-1729753202. Without access to changing GitHub CI action myself, it is impossible to test, so if you don't have time to work on it, then I can just drop the CI part of this PR. My main motivation was to prevent style-related CI fails on PRs and the local use of pre-commit does it.

Reply:
>  I don't think I can implement that with my current level of access in the Pymoca organization. 

I thought I made you maintainer, I'll see if there's a higher level

Reply:
> I did not try to see if we can omit the rev.

I confirmed that the rev key is required. pre-commit.ci has an option that will automatically update the revs to latest and commit the update on a weekly basis. It also has an option to automatically fix files that aren't black conformant, but I'm not thrilled with the idea of polluting the history with bot commits. If we scale up to more collaborators, it might make sense to revisit this.